lib/events/athena/athena.go

```diff
@@ -0,0 +1,355 @@
+// Right now the only way to set config is via url params.
+type Config struct {
+	// Region is where Athena, SQS and SNS lives (required).
+	Region string
+
+	// Publisher settings.
+
+	// TopicARN where to emit events in SNS (required).
+	TopicARN string
+	// LargeEventsS3 is location on S3 where temporary large events (>256KB)
+	// are stored before converting it to Parquet and moving to long term
+	// storage (required).
+	LargeEventsS3 string
+
+	// Query settings.
+
+	// Database is name of Glue Database that Athena will query against (required).
+	Database string
+	// TableName is name of Glue Table that Athena will query against (required).
+	TableName string
+	// LocationS3 is location on S3 where Parquet files partitioned by date are
+	// stored (required).
+	LocationS3 string
+	// QueryResultsS3 is location on S3 where Athena stored query results (optional).
+	// Default results path can be defined by in workgroup settings.
+	QueryResultsS3 string
+	// Workgroup is Glue workgroup where Athena queries are executed (optional).
+	Workgroup string
+	// GetQueryResultsSleepTime is used to define how long query will wait before
+	// checking again for results status if previous status was not ready (optional).
+	GetQueryResultsSleepTime time.Duration
+	// LimiterRate defines rate at which search_event rate limiter is filled (optional).
+	LimiterRate float64
+	// LimiterBurst defines rate limit bucket capacity (optional).
+	LimiterBurst int
+
+	// Batcher settings.
+
+	// QueueUrl is URL of SQS, which is set as subscriber to SNS topic (required).
+	QueueUrl string
+	// BatchMaxItems defines how many items can be stored in single Parquet
+	// batch (optional).
+	// It's soft limit.
+	BatchMaxItems int
+	// BatchMaxInterval defined interval at which parquet files will be created (optional).
+	BatchMaxInterval time.Duration
+
+	// Clock is a clock interface, used in tests.
+	Clock clockwork.Clock
+	// UIDGenerator is unique ID generator.
+	UIDGenerator utils.UID
+
+	// TODO(tobiaszheller): add FIPS config in later phase.
+}
+
+// CheckAndSetDefaults is a helper returns an error if the supplied configuration
+// is not enough to connect to SNS
+func (cfg *Config) CheckAndSetDefaults() error {
+	const glueNameMaxLen = 255
+	if cfg.Database == "" {
+		return trace.BadParameter("Database is not specified")
+	}
+	if len(cfg.Database) > glueNameMaxLen {
+		return trace.BadParameter("Database name too long")
+	}
+	if !isAlphanumericOrUnderscore(cfg.Database) {
+		return trace.BadParameter("Database name can contains only alphanumeric or underscore characters")
+	}
+
+	if cfg.TableName == "" {
+		return trace.BadParameter("TableName is not specified")
+	}
+	if len(cfg.TableName) > glueNameMaxLen {
+		return trace.BadParameter("TableName too long")
+	}
+	// TableName is appended directly to athena query. That's why we put extra care
+	// that no weird chars are passed here.
+	if !isAlphanumericOrUnderscore(cfg.TableName) {
```

Comment:
it's AWS limitation: https://docs.aws.amazon.com/athena/latest/ug/tables-databases-columns-names.html
I wanted to add that validation explicitly to avoid cases when someone tries to do sql injection via table name, because in athena query builder you cannot pass table name as query param.